libs/vkd3d/command.c

```diff
@@ -6089,70 +6090,128 @@ static void d3d12_command_list_transition_image_layout(struct d3d12_command_list
         assert(d3d12_resource_is_buffer(dst_resource));
         assert(d3d12_resource_is_texture(src_resource));
 
-        if (!(src_format = vkd3d_format_from_d3d12_resource_desc(list->device,
-                &src_resource->desc, DXGI_FORMAT_UNKNOWN)))
+        if (!(out->src_format = vkd3d_format_from_d3d12_resource_desc(list->device,
+                      &src_resource->desc, DXGI_FORMAT_UNKNOWN)))
         {
             WARN("Invalid format %#x.\n", dst->PlacedFootprint.Footprint.Format);
-            return;
+            return false;
         }
 
-        if (!(dst_format = vkd3d_get_format(list->device, dst->PlacedFootprint.Footprint.Format,
-                true)))
+        if (!(out->dst_format = vkd3d_get_format(list->device, dst->PlacedFootprint.Footprint.Format,
+                      true)))
         {
             WARN("Invalid format %#x.\n", dst->PlacedFootprint.Footprint.Format);
-            return;
+            return false;
         }
 
-        vk_image_buffer_copy_from_d3d12(&buffer_image_copy, &dst->PlacedFootprint,
-                src->SubresourceIndex, &src_resource->desc, src_format, dst_format,
+        vk_image_buffer_copy_from_d3d12(&out->copy.buffer_image, &dst->PlacedFootprint,
+                src->SubresourceIndex, &src_resource->desc, out->src_format, out->dst_format,
                 src_box, dst_x, dst_y, dst_z);
-        buffer_image_copy.bufferOffset += dst_resource->mem.offset;
+        out->copy.buffer_image.bufferOffset += dst_resource->mem.offset;
+
+        out->vk_layout = d3d12_resource_pick_layout(src_resource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL);
+    }
+    else if (src->Type == D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT && dst->Type == D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX)
+    {
+        assert(d3d12_resource_is_texture(dst_resource));
+        assert(d3d12_resource_is_buffer(src_resource));
+
+        if (!(out->dst_format = vkd3d_format_from_d3d12_resource_desc(list->device,
+                      &dst_resource->desc, DXGI_FORMAT_UNKNOWN)))
+        {
+            WARN("Invalid format %#x.\n", dst_resource->desc.Format);
+            return false;
+        }
 
-        vk_layout = d3d12_resource_pick_layout(src_resource, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL);
+        if (!(out->src_format = vkd3d_get_format(list->device, src->PlacedFootprint.Footprint.Format,
+                      true)))
+        {
+            WARN("Invalid format %#x.\n", src->PlacedFootprint.Footprint.Format);
+            return false;
+        }
+
+        vk_buffer_image_copy_from_d3d12(&out->copy.buffer_image, &src->PlacedFootprint,
+                dst->SubresourceIndex, &dst_resource->desc, out->src_format, out->dst_format, src_box, dst_x,
+                dst_y, dst_z);
+        out->copy.buffer_image.bufferOffset += src_resource->mem.offset;
+
+        out->vk_layout = d3d12_resource_pick_layout(dst_resource, VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL);
```

Comment:
Writes_full_subresource is not set here. I don't think the original code did this either, but it should be added I think.